app/controllers/groups_controller.rb

```diff
@@ -0,0 +1,77 @@
+class GroupsController < ApplicationController
+  load_and_authorize_resource
+
+  before_action :find_user
+  before_action :find_group, only: %i[show edit update destroy]
+
+  def index
+    @groups = @user.groups.all
+  end
+
+  def show; end
+
+  def new
+    @group = Group.new
+  end
+
+  def create
+    @group = Group.new(group_params)
+    @group.user = @user
+    if @group.save
+      redirect_to groups_path, notice: 'Group created successfully'
+    else
+      flash.now[:alert] = @group.errors.full_messages.first if @group.errors.any?
+      render :new, status: 400
+    end
+  end
+
+  def edit; end
+
+  def update
+    if @group.update(group_params)
+      redirect_to group_path(id: @group.id), notice: 'Group updated successfully'
+    else
+      flash.now[:alert] = @group.errors.full_messages.first if @group.errors.any?
+      render :edit, status: 400
+    end
+  end
+
+  # rubocop:disable Lint/UselessAssignment
+  def destroy
+    if can? :edit, @group
+      @group_expenses = GroupExpense.where(group_id: @group.id)
+      @group_expenses.each do |group_expense|
+        expense_id = group_expense.expense_id
+        group_expense.destroy
+        expense = Expense.delete(expense_id)
+      end
+      if @group.destroy
+        redirect_to groups_path, notice: 'Groups was deleted successfully'
+      else
+        flash.now[:alert] = @group.errors.full_messages.first if @group.errors.any?
+        render :index, status: 400
+      end
+    else
+      flash[:alert] = 'Un Authorized'
+      redirect_to groups_path
+    end
+  end
+  # rubocop:enable Lint/UselessAssignment
+
```

Comment:
- [ ] It is always a good practice to make sure you fix all linter errors instead of disabling them, A linter is great for identifying errors when you use standard rules. Remember, a linter analyzes your code for stylistic and programming errors against the rules it knows. 💯 😉